connectors/src/connectors/slack/lib/slack_client.ts

```diff
@@ -104,10 +104,64 @@ export async function getSlackClient(
   return proxied;
 }
 
-export async function getSlackUserInfo(slackClient: WebClient, userId: string) {
-  return slackClient.users.info({
-    user: userId,
-  });
+export type SlackUserInfo = {
+  email: string | null;
+  is_bot: boolean;
+  display_name?: string;
+  real_name?: string;
+  is_restricted: boolean;
+  is_stranger: boolean;
+  is_ultra_restricted: boolean;
+  teamId: string | null;
+  tz: string | null;
+  image_512: string | null;
+};
+
+export async function getSlackUserInfo(
+  slackClient: WebClient,
+  userId: string
+): Promise<SlackUserInfo> {
+  const res = await slackClient.users.info({ user: userId });
+
+  if (!res.ok) {
+    throw res.error;
+  }
+
+  return {
+    is_bot: false,
+    email: res.user?.profile?.email || null,
+    display_name: res.user?.profile?.display_name,
+    real_name: res.user?.profile?.real_name,
+    is_restricted: res.user?.is_restricted || false,
+    is_stranger: res.user?.is_stranger || false,
+    is_ultra_restricted: res.user?.is_ultra_restricted || false,
+    teamId: res.user?.team_id || null,
+    tz: res.user?.tz || null,
+    image_512: res.user?.profile?.image_512 || null,
+  };
+}
+
+export async function getSlackBotInfo(
+  slackClient: WebClient,
+  botId: string
+): Promise<SlackUserInfo> {
+  const slackBot = await slackClient.bots.info({ bot: botId });
+  if (slackBot.error) {
+    throw slackBot.error;
+  }
+
+  return {
+    display_name: slackBot.bot?.name,
+    real_name: slackBot.bot?.name,
+    email: null,
+    image_512: slackBot.bot?.icons?.image_72 || null,
+    tz: null,
+    is_restricted: false,
```

Comment:
Are you sure about that?

can’t there be bots that are not fully vetted by the org?

Reply:
For the record:

Here is my understanding of bots permissions system:
- They can be installed via a Slack app (like our Dust Slack bot is installed on Clay's workspace), by admins of the Slack workspace or all members of the slack workspace depending on the Slack workspace settings.
- Workflow automation can send messages on channel as a bot. For example, a workflow can send a message mentioning @.Dust when someones sets a specific emoji on a specific channel. That would allow anyone to effectively trigger our @.Dust bot, the content of the message is controlled by the person writing the workflow. Is there security hole to that? Are there workflow that could trigger an extra mentions? I don't know, but we only accept one mention, or we drop them all. Can we rely on that as a security measure? I don't think so.
- External users of the workspace (like employees of Clay on our #clay-dust channel) can be invited with or without the right to trigger workflows on Slack (see screenshots), so to me it brings us back to: it depends on the permissions you give them. Might be naive but is there another solution to it?

![Screenshot 2024-03-16 at 09 06 16](https://github.com/dust-tt/dust/assets/358965/892ff61f-a5f4-4682-947f-8d0ef45a84c0)